src/main/java/org/hiero/sketch/spreadsheet/Bucket.java

```diff
@@ -0,0 +1,43 @@
+package org.hiero.sketch.spreadsheet;
+import java.util.*;
+/**
+ * A Bucket to be used as part of a histogram.
+ */
+public class Bucket<T> {
```

Comment:
I think that this class is too general and not general enough at the same time.
- it's too general since it uses a generic T; all our datatypes are convertible to double, so probably we'll use doubles for all histograms
- it's not general enough since it does not support multi-dimensional histograms - there's just a left and a right boundary.
I propose to go the opposite way about this: instead of designing the most general histogram, let's build first a 1-D histogram of doubles with equally-spaced buckets, where the left is always inclusive except the rightmost bucket where the right is also inclusive. Then we build a 2-D histogram class, and then we see how we can generalize by merging the common code. So I would start by designing a OneDimensionalHistogram class, which maintains internally an array of buckets.

Please note that all histogram classes have to maintain a special bucket for "missing" data.